maven-plugin/src/main/java/io/apicurio/codegen/maven/GenerateCodeMojo.java

```diff
@@ -0,0 +1,106 @@
+package io.apicurio.codegen.maven;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Enumeration;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import io.apicurio.hub.api.codegen.JaxRsProjectSettings;
+import io.apicurio.hub.api.codegen.OpenApi2JaxRs;
+
+/**
+ * The main generate code mojo implementation.
+ * 
+ * @author eric.wittmann@gmail.com
+ */
+@Mojo(name = "generate")
+public class GenerateCodeMojo extends AbstractMojo {
+
+    @Parameter(required = true)
+    JaxRsProjectSettings projectSettings;
+
+    @Parameter(required = true)
+    File inputSpec;
+
+    @Parameter(defaultValue = "${project.build.directory}/generated-sources/jaxrs")
+    File outputDir;
+
+    /**
+     * @see org.apache.maven.plugin.Mojo#execute()
+     */
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        getLog().info("Generating JAX-RS interfaces and beans from: " + inputSpec);
+        if (inputSpec == null || !inputSpec.isFile()) {
+            throw new MojoFailureException("Input spec not found.");
+        }
+
+        if (outputDir.isFile()) {
+            throw new MojoFailureException(
+                    "Output directory is unexpectedly a file (should be a directory or non-existent).");
+        }
+
+        if (!outputDir.exists()) {
+            outputDir.mkdirs();
+        }
+
+        // Generate code - output a ZIP file.
+        File zipFile = new File(outputDir, "generated-code.zip");
```

Comment:
I "think" that it might be cleaner to generate the `zip` file in a temporary folder, in case the build breaks you end up with a `zip` file along with (possible) sources.
Something "close enough" can be done by having `zipFile.delete()` in a `finally` block 🙂

Reply:
I disagree with using a temporary folder (assuming you meant a system temp folder).  I think the `target` directory is specifically meant for these sorts of files.

However, you're totally right about using a finally block.  That was a bad oversight on my part.  :)  Adding that now.